docs/proposals/resource-metrics-api.md

```diff
@@ -0,0 +1,177 @@
+of the scope of the MVP version.
+
+#### Custom metrics in HPA
+
+HPA requires the latest value of application level metrics.
+
+The design of the pipeline for collecting application level metrics should
+be revisited and it's not clear whether application level metrics should be
+available in API server so the use case initially won't be supported.
+
+#### Ubernetes
+
+Ubernetes might want to consider cluster-level usage (in addition to cluster-level request)
+of running pods when choosing where to schedule new pods. Although Ubernetes is still in design,
+we expect the metrics API described here to be sufficient. Cluster-level usage can be
+obtained by summing over usage of all nodes in the cluster.
+
+#### kubectl top
+
+This feature is not yet specified/implemented although it seems reasonable to support the following options:
+- ```kubectl top node <id>``` - list all pods running on the given node accompanied with theirs cpu/mem usage
+- ```kubectl top pod <id>``` - print cpu/mem usage stats for the given pod
+- ```kubectl top pod --selector=<label-query>``` - list all pods defined by the label query accompanied with their cpu/mem usage
+
+Since this feature has not been fully specified yet it will be not supported initally in the API although
+it will be probably possible to provide a reasonable implementation of the feature anyway.
+
+#### Kubernetes dashboard
+
+[Kubernetes dashboard](https://github.com/kubernetes/dashboard) in order to draw graps requires resource usage
+in timeseries format from relatively long period of time. The aggreations should be also possible on various levels
+including replication controllers, deployments, services, etc.
+
+Since the use case is complicated it will not be supported initally in the API and they will query Heapster
+directly using some custom API there.
+
+## Proposed API
+
+Initially the metrics API will be in a separate [API group](api-group.md) called ```metrics```.
+Later if we decided to have Node and Pod in different API groups also
+NodeMetrics and PodMetrics should be in different API groups.
+
+#### Schema
+
+The proposed schema is as follow. Each top-level object has `TypeMeta` and `ObjectMeta` fields
+to be compatible with Kubernetes API standards.
+
+```go
+type NodeMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Usage v1.ResourceList
+}
+
+type PodMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Containers []ContainerMetrics
+}
+
+type ContainerMetrics struct {
+  Name  string
+  Usage v1.ResourceList
```

Comment:
Either 'everything we have' (if we collected only memory, we will return only memory) or 'all or nothing' (if we collected only memory, we will respond with 404). I don't have strong opinion which one is better. WDYT?